### PR TITLE
Add sparse convertion api and sparse creation api

### PR DESCRIPTION
**Pull request by zkh2016**
### PR types

Others
### PR changes

Others
### Describe

1. eager.Tensor中添加to_sparse_coo、to_sparse_csr、to_dense三个方法，用来dense和sparse互转使用。
2. eager中添加sparse_coo_tensor和sparse_csr_tensor两个API，用来创建SparseCooTensor和SparseCsrTensor。

### CODE REVIEW

File: paddle/phi/api/lib/CMakeLists.txt [hunk 1]
```diff
@@ -143,7 +143,7 @@ cc_library(phi_data_transform SRCS data_transform.cc DEPS phi_tensor_raw transfe
 cc_library(api_custom_impl SRCS api_custom_impl.cc DEPS phi_tensor_raw phi kernel_dispatch api_gen_utils phi_data_transform)
 cc_library(sparse_api_custom_impl SRCS sparse_api_custom_impl.cc DEPS phi_tensor_raw phi kernel_dispatch api_gen_utils phi_data_transform)
 
-cc_library(phi_function_api SRCS ${api_source_file} DEPS phi_tensor_raw phi kernel_dispatch api_gen_utils phi_data_transform api_custom_impl)
+cc_library(phi_function_api SRCS ${api_source_file} DEPS phi_tensor_raw phi kernel_dispatch api_gen_utils phi_data_transform api_custom_impl sparse_api_custom_impl)
```
Comment (zyfncg) [inline paddle/phi/api/lib/CMakeLists.txt]: 这里的依赖应该不需要加，普通的前向API不会依赖自定义的sparse_api

File: paddle/phi/api/include/tensor.h [hunk 2]
```diff
@@ -520,6 +520,30 @@ class PADDLE_API Tensor final {
...
+   */
+  Tensor to_sparse_csr();
+
+  /**
+   * @brief Convert SparseCooTensor or SparseCsrTensor to DenseTensor
+   *
+   * @return Tensor
+   */
+  Tensor to_dense();
+
```
Comment (zyfncg) [inline paddle/phi/api/include/tensor.h]: 这几个方法标记成 const？

File: paddle/phi/api/lib/CMakeLists.txt (hunk 1 above)
Comment (zkh2016) [reply paddle/phi/api/lib/CMakeLists.txt]: 已去掉，改成在编译tensor_methord.cc地方添加依赖。

File: paddle/phi/api/include/tensor.h (hunk 2 above)
Comment (zkh2016) [reply paddle/phi/api/include/tensor.h]: done

**Review by jim19930609 — APPROVED** [review]

**Review by zyfncg — APPROVED** [review]

File: paddle/fluid/pybind/eager_method.cc [hunk 3]
```diff
@@ -810,6 +810,49 @@ static PyObject* tensor_method_is_sparse_csr(TensorObject* self, PyObject* args,
...
+  egr::EagerUtils::autograd_meta(&coo_tensor)
+      ->SetStopGradient(
+          egr::EagerUtils::autograd_meta(&self->tensor)->StopGradient());
+  egr::EagerUtils::autograd_meta(&coo_tensor)
+      ->SetPersistable(
+          egr::EagerUtils::autograd_meta(&(self->tensor))->Persistable());
+  return ToPyObject(coo_tensor);
+  EAGER_CATCH_AND_THROW_RETURN_NULL
+}
+
```
Comment (OccupyMars2030) [inline paddle/fluid/pybind/eager_method.cc:827]: I cannot  find this method "tensor_method_to_sparse_coo"  in 
 paddle\fluid\pybind\eager_method.cc  on  the  "develop" branch.

File: paddle/fluid/pybind/eager_method.cc [hunk 4]
```diff
@@ -810,6 +810,49 @@ static PyObject* tensor_method_is_sparse_csr(TensorObject* self, PyObject* args,
...
+  egr::EagerUtils::autograd_meta(&dense_tensor)
+      ->SetStopGradient(
+          egr::EagerUtils::autograd_meta(&self->tensor)->StopGradient());
+  egr::EagerUtils::autograd_meta(&dense_tensor)
+      ->SetPersistable(
+          egr::EagerUtils::autograd_meta(&(self->tensor))->Persistable());
+  return ToPyObject(dense_tensor);
+  EAGER_CATCH_AND_THROW_RETURN_NULL
+}
+
```
Comment (OccupyMars2030) [inline paddle/fluid/pybind/eager_method.cc:855]: Now, I cannot find "tensor_method_to_dense" in paddle\fluid\pybind\eager_method.cc  on 
the "develop" branch

File: paddle/fluid/pybind/eager_method.cc (hunk 3 above)
Comment (zkh2016) [reply paddle/fluid/pybind/eager_method.cc:827]: 'to_sparse_coo' has been move to [varbase_patch_methods.py](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/dygraph/varbase_patch_methods.py#L967)

File: paddle/fluid/pybind/eager_method.cc (hunk 4 above)
Comment (zkh2016) [reply paddle/fluid/pybind/eager_method.cc:855]: ‘to_dense' has been move to [varbase_patch_methods.py](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/dygraph/varbase_patch_methods.py#L934)

**Pull request merged**